### PR TITLE
LocalWorker: Fix nr threads to 1

### PR DESCRIPTION
**Pull request by mtrofin**
From discussion on PR #75

### CODE REVIEW

File: compiler_opt/distributed/local/local_worker_manager.py [hunk 1]
```diff
@@ -120,13 +120,16 @@ def __init__(self):
...
+      # to handle high priority requests. The expectation is that the user
+      # achieves concurrency through multiprocessing, not multithreading.
       self._process = multiprocessing.Process(
           target=functools.partial(
               _run,
               worker_class=cls,
               in_q=self._send,
               out_q=self._receive,
-              threads=pool_threads,
+              threads=1,
```
Comment (yundiqian) [inline compiler_opt/distributed/local/local_worker_manager.py:132]: just have max_workers=1 in line 76?

**Review by yundiqian — COMMENTED** [review]
delete `import gin`

File: compiler_opt/distributed/local/local_worker_manager.py (hunk 1 above)
Comment (mtrofin) [reply compiler_opt/distributed/local/local_worker_manager.py]: yup

**Comment by boomanaiden154** [comment]
I've found that setting the number of threads in the `ThreadPoolExecutor` to something greater than 1 helps with increasing CPU utilization, but that might only be when setting the `num_workers` CLI option to something significantly less than the number of threads available on the system. If set equal to the number of threads in the system though (the default), it is fine if I recall correctly. If it becomes an issue I can always just do some benchmarking and adjust the variable later.

**Pull request merged**